### PR TITLE
feat(CLI): First iteration of support for `verbose` mode in `deploy`

### PR DESCRIPTION
**Pull request by pgrzesik**
Addresses: #9860

First iteration of verbose mode in `deploy` for modern logs, it will require adjustments as current output is definitely not the final version.

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/extendedValidate.js [hunk 1]
```diff
@@ -46,13 +46,14 @@ module.exports = {
...
-                `WARNING: Function ${functionName} has timeout of ${functionObject.timeout} `,
+              const warningText = [
+                `Function ${functionName} has timeout of ${functionObject.timeout} `,
                 "seconds, however, it's attached to API Gateway so it's automatically ",
                 'limited to 30 seconds.',
               ].join('');
+              log.warning(warningText);
 
-              legacy.log(warnMessage);
+              legacy.log(`WARNING: ${warningText}`);
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/extendedValidate.js:56]: Just a note: This change looks ok. Still, it seems out of the scope of this PR (warnings were supposed to be covered in later turn)

File: lib/plugins/aws/deploy/lib/uploadArtifacts.js [hunk 2]
```diff
@@ -35,9 +35,12 @@ module.exports = {
     if (artifactFilePaths.length === 1) {
       const stats = await this.getFileStats(artifactFilePaths[0]);
       progress.get('main').notice(`Uploading (${filesize(stats.size)})`);
+      // log.info(`Uploading (${filesize(stats.size)})`);
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/uploadArtifacts.js:38]: Let's not add commented out code

File: lib/plugins/aws/info/display.js [hunk 3]
```diff
@@ -172,6 +172,15 @@ module.exports = {
       legacy.consoleLog(message);
     }
 
+    if (isVerboseMode && this.gatheredData.outputs) {
```
Comment (medikoo) [inline lib/plugins/aws/info/display.js:175]: `this.gatheredData.outputs` is guaranteed to exist. I think we need to confirm on length

File: lib/plugins/aws/lib/monitorStack.js [hunk 4]
```diff
@@ -131,6 +136,7 @@ module.exports = {
               ) {
                 // empty console.log for a prettier output
                 if (!this.options.verbose) legacy.consoleLog('');
+                // TODO: CloudFormation error handling to be done as a separate PR
```
Comment (medikoo) [inline lib/plugins/aws/lib/monitorStack.js:139]: I think we don't need that comment (we track that in our initiative)

File: lib/plugins/package/lib/packageService.js [hunk 5]
```diff
@@ -69,14 +69,16 @@ module.exports = {
   async packageService() {
     legacy.log('Packaging service...');
     progress.get('main').notice('Packaging');
+    log.info('\nPackaging');
```
Comment (medikoo) [inline lib/plugins/package/lib/packageService.js:72]: It's safer to do:

```javascript
log.info();
log.info('Packaging');
```

Thing is, that if e.g. later we will decide to prefix verbose logs (e.g. with timestamp), then for `log.info('\nPackaging')` we will have timestamp and text after the new line

File: lib/plugins/package/lib/packageService.js [hunk 6]
```diff
@@ -204,6 +206,7 @@ module.exports = {
       await this.excludeDevDependencies({
         exclude: this.getExcludes([], true),
         include: this.getIncludes(),
+        excludingFor: 'service package',
```
Comment (medikoo) [inline lib/plugins/package/lib/packageService.js:209]: I wonder whether `contextName` property name wouldn't be better

**Review by medikoo — CHANGES_REQUESTED** [review]
Thanks @pgrzesik, in general, it looks good. I have just a few remarks

File: lib/plugins/package/lib/packageService.js (hunk 5 above)
Comment (pgrzesik) [reply lib/plugins/package/lib/packageService.js:72]: good call :+1:

File: lib/plugins/aws/info/display.js [hunk 7]
```diff
@@ -172,6 +172,15 @@ module.exports = {
       legacy.consoleLog(message);
     }
 
+    if (isVerboseMode && this.gatheredData.outputs.length > 0) {
```
Comment (medikoo) [inline lib/plugins/aws/info/display.js:175]: nit: `this.gatheredData.outputs.length` is equivalent to `this.gatheredData.outputs.length > 0`

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**